lib/media/drm_engine.js

```diff
@@ -413,6 +413,57 @@ shaka.media.DrmEngine = class {
     /** @type {!Map.<string, MediaKeySystemConfiguration>} */
     let configsByKeySystem;
 
+    /**
+     * Expand robustness into multiple drm infos if multiple video robustness
+     * levels were provided.
+     *
+     * robustness can be either a single item as a string or multiple items as
+     * an array of strings.
+     */
+    const expandRobustness = (drmInfos, robustnessType) => {
+      const newDrmInfos = [];
+      for (const drmInfo of drmInfos) {
+        let items = drmInfo[robustnessType] ||
+            (this.config_.advanced &&
+            this.config_.advanced[drmInfo.keySystem] &&
+            this.config_.advanced[drmInfo.keySystem][robustnessType]) || '';
+        if (typeof items === 'string') {
```

Comment:
This is necessary because as we loop through the drmInfos and expand the robustness array into an array of DrmInfo with a string robustness, those updated items will end up back in here as we loop through subsequent variants that may be pointing at the same video or audio streams.

Reply:
Maybe this is worth a comment? It tripped me up at first when I was writing some tests because I didn't have this clause.